imports/ui/PreviewCard.jsx

```diff
@@ -0,0 +1,216 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import './css/PreviewCard.css';
+import Gauge from 'react-summary-gauge-2';
+import { getGaugeValues } from './js/CourseCard.js';
+import Review from './Review.jsx';
+
+/*
+  Filtered Result Component.
+*/
+
+export default class PreviewCard extends Component {
+  constructor(props) {
+    super(props);
+    // set gauge values
+    this.state = {
+      id: this.props.course._id,
+      rating: this.props.course.classRating,
+      ratingColor: "E64458",
+      diff: this.props.course.classDifficulty,
+      diffColor: "E64458",
+      workload: this.props.course.classWorkload,
+      workloadColor: "E64458",
+      topReview: {},
+      numReviews: 0
+    };
+
+    this.updateColors = this.updateColors.bind(this);
+    this.updateTopReview = this.updateTopReview.bind(this);
+    this.updateGauges = this.updateGauges.bind(this);
+
+  }
+
+  componentDidMount() {
+    this.updateGauges();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps != this.props) {
+      this.updateGauges();
+    }
+  }
+
+  updateGauges() {
+    this.setState({
+      id: this.props.course._id,
+      rating: this.props.course.classRating == null ? "-" : this.props.course.classRating,
+      diff: this.props.course.classDifficulty == null ? "-" : this.props.course.classDifficulty,
+      workload: this.props.course.classWorkload == null ? "-" : this.props.course.classWorkload,
+    }, () => this.updateColors(), this.updateTopReview());
```

Comment:
`this.updateTopReview()` will be ignored since setState only accepts 2 arguments

Reply:
Good call, we'll fix.

Reply:
Fixed in latest commit.